src/main/java/com/synopsys/integration/detect/configuration/DetectableOptionFactory.java

```diff
@@ -142,6 +171,11 @@ public GradleInspectorOptions createGradleInspectorOptions() {
         List<String> includedConfigurationNames = getValue(DetectProperties.DETECT_GRADLE_INCLUDED_CONFIGURATIONS);
         String customRepository = ArtifactoryConstants.GRADLE_INSPECTOR_MAVEN_REPO;
         boolean includeUnresolvedConfigurations = getValue(DetectProperties.DETECT_GRADLE_INCLUDE_UNRESOLVED_CONFIGURATIONS);
+        if (detectConfiguration.wasPropertyProvided(DetectProperties.DETECT_GRADLE_CONFIGURATION_TYPES_EXCLUDED.getProperty())) {
+            List<GradleConfigurationType> excludedConfigurationTypes = PropertyConfigUtils.getNoneList(detectConfiguration, DetectProperties.DETECT_GRADLE_CONFIGURATION_TYPES_EXCLUDED.getProperty()).representedValues();
+            ExcludedDependencyTypeFilter<GradleConfigurationType> dependencyTypeFilter = new ExcludedDependencyTypeFilter<>(excludedConfigurationTypes);
```

Comment:
I fell that ExcludedDependencyTypeFilter does not adequately describe the type. Something slightly more generic would be preferred. ExcludableEnumFilter, FilterableEnum, ReportableEnumFilter, ReportFilter, EnumReportFilter.